tests/integration/test_agent.py

```diff
@@ -40,7 +40,7 @@ def test_write_simple_script():
     task = "Write a shell script 'hello.sh' that prints 'hello'. Do not ask me for confirmation at any point."
     final_state: State = asyncio.run(main(task, exit_on_message=True))
     assert final_state.agent_state == AgentState.STOPPED
-    assert final_state.error is None
+    assert final_state.eval_error is None
```

Comment:
As you can see here, we do use it in testing, so it's not only used by evaluation. Also "eval_error" sounds a bit weird to me - the error itself has nothing to do with evaluation.

Reply:
I'd concurr. Only the circumstances make it "related" to evaluations, but the code doesn't know that, only the user (you 😁 ).

Reply:
The point here is we never clear it or even read it in the backend. The fact that we never clear it made me clear it when the agent wasn't in error anymore... Or to put it differently, this is a one-and-done value, the rest of the application continues after an error, the agent changes state, even immediately after error (it becomes stopped or paused), but this value is a special value that is supposed to be 'error' when the state isn't in error anymore. The reason for that is eval: the only kind of use that needs to know that the agent _was_ in error state in the past.

Maybe there is some alternative, so we don't let it stay forever "error"?

Reply:
> The point here is we never clear it or even read it in the backend. The fact that we never clear it made me clear it when the agent wasn't in error anymore... 

I do notice this as well. I propose we fix the real bug (clear the error flag once we recover) instead.

From evaluation perspective, does it matter that the agent "once encountered an error"? It seems to me evaluation only cares about final state.

Reply:
If `last_error` works, at least for now, I think it's better, the name implies to me that it can change in the future and indeed it would (at the next error), and that's okay.

At some point, I cleared it when the agent changed state away from ERROR, and that caused a problem with eval. I think what happened was the evaluations didn't get anymore the error information they add to the report. (e.g. "stuck in a loop"). In that sense, evals still need a little bit of a past state of the agent, when the backend has already changed to STOPPED or something. I can imagine more complex ways to implement this, but the simplest for now is just to name it so it doesn't confuse us...